src/components/reusable/daterangepicker/daterangepicker.ts

```diff
@@ -0,0 +1,289 @@
+ * @prop {string} maxDate - Maximum date in YYYY-MM-DD format. If the value isn't a possible date string in the format, then the element has no maximum date value
+ * @slot unnamed - Slot for label text.
+ */
+@customElement('kyn-date-range-picker')
+export class DateRangePicker extends LitElement {
+  static override styles = DateRangePickerScss;
+  /**
+   * Associate the component with forms.
+   * @ignore
+   */
+  static formAssociated = true;
+
+  /** Regex format for minDate / maxDate
+   * @ignore
+   */
+  @state()
+  regexDateFormat = /^\d{4}-\d{2}-\d{2}$/;
+
+  /**
+   * Attached internals for form association.
+   * @ignore
+   */
+  @state()
+  internals = this.attachInternals();
+
+  /** Optional text beneath the input. */
+  @property({ type: String })
+  caption = '';
+
+  /** Datepicker size. "sm", "md", or "lg". */
+  @property({ type: String })
+  size = 'md';
+
+  /** Datepicker Start date in YYYY-MM-DD format. */
+  @property({ type: String })
+  startDate = '';
+
+  /** Datepicker End date in YYYY-MM-DD format. */
+  @property({ type: String })
+  endDate = '';
+
+  /** Datepicker name. Required prop. as there could many fields into single form*/
+  @property({ type: String })
+  name = '';
+
+  /** Makes the date required. */
+  @property({ type: Boolean })
+  required = false;
+
+  /** Date disabled state. */
+  @property({ type: Boolean })
+  disabled = false;
+
+  /** Date invalid text. */
+  @property({ type: String })
+  invalidText = '';
+
+  /** Date warning text */
+  @property({ type: String })
+  warnText = '';
+
+  /** Maximum date in YYYY-MM-DD format.
+   * If the value isn't a possible date string in the format, then the element has no maximum date value
+   */
+  @property({ type: String })
+  maxDate = '';
+
+  /** Minimum date in YYYY-MM-DD format,
+   * If the value isn't a possible date string in the format, then the element has no minimum date value.
+   */
+  @property({ type: String })
+  minDate = '';
+
+  /** Specifies the granularity that the value must adhere to, or the special value any,
+   * For date inputs, the value of step is given in days; and is treated as a number of milliseconds equal to 86,400,000 times the step value.
+   * The default value of step is 1, indicating 1 day.*/
+  @property({ type: String })
+  step = '';
+
+  override render() {
+    return html`
+      <label
+        class="datepicker-label-text"
+        for=${this.name}
+        ?disabled=${this.disabled}
+      >
+        ${this.required ? html`<span class="required">*</span>` : null}
+        <slot></slot>
+      </label>
+
+      <div class="input-wrapper">
+        <input
+          class="${classMap({
+            'size--sm': this.size === 'sm',
+            'size--lg': this.size === 'lg',
+          })}"
+          type="date"
+          id=${this.name ? `${this.name}-start` : 'date-range-start'}
+          name=${this.name}
```

Comment:
'name' value should be the same as 'id' here. The fallback 'date-range-start' is not unique however, if there's more than one on a page. Fallback value is not unique if multiple on page.